### PR TITLE
Produce specialized conformances to AnyObject for bound generic types

### PR DESCRIPTION
**Pull request by swiftix**
Produce specialized conformances to AnyObject for bound generic types.

Fixes rdar://31910351 (SR-4750)

### CODE REVIEW

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@slavapestov Does it look OK? I need to submit for 4.0 as well.

File: lib/AST/SubstitutionMap.cpp [hunk 1]
```diff
@@ -133,7 +135,6 @@ Type SubstitutionMap::lookupSubstitution(CanSubstitutableType type) const {
   // The generic parameter may have been made concrete by the generic signature,
   // substitute into the concrete type.
   ModuleDecl &anyModule = *genericParam->getASTContext().getStdlibModule();
-  auto genericSig = getGenericSignature();
```
Comment (slavapestov) [inline lib/AST/SubstitutionMap.cpp]: Does this compile? Where is genericSig defined?

**Comment by slavapestov** [comment]
Looks good, but the commit message is out of date. Were you unable to reduce a test case?

File: lib/AST/SubstitutionMap.cpp (hunk 1 above)
Comment (swiftix) [inline lib/AST/SubstitutionMap.cpp]: It is defined above, I moved it higher. Look for line 119.

**Comment by swiftix** [comment]
No, I haven't reduced the test case. Will try now.

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Pull request merged**